SDWebImage/Core/SDImageCoderHelper.m

```diff
@@ -371,18 +371,18 @@ + (CGImageRef)CGImageCreateScaled:(CGImageRef)cgImage size:(CGSize)size {
         .renderingIntent = CGImageGetRenderingIntent(cgImage)
     };
     
-    vImage_Error a_ret = vImageBuffer_InitWithCGImage(&input_buffer, &format, NULL, cgImage, kvImageNoFlags);
+    vImage_Error a_ret = vImageBuffer_InitWithCGImage(&input_buffer, &format, NULL, cgImage, kvImageNoAllocate);
```

Comment:
BUG:

When use `kvImageNoAllocate`, vImage can not automatic convert to the format we want.

For example, if the input CGImage is 24bit, non-alpha image, we requests 32bit, alpha image, using `kvImageNoAllocate` will return failure

So this should not use  :)